frontend/src/components/molecules/DetailView.tsx

```diff
@@ -39,8 +46,15 @@ const DetailView: React.FC = () => {
     setDevelopmentAreas(await getDevelopmentArea(selectedNode.id));
   };
 
-  const expandConnection = (connection: Node) => {
+  const loadObjectPropertyDefinition = async () => {
+    if (!selectedPredicate) return;
+    setObjectAnnotations(await getAnnotations(selectedPredicate[1]));
+  };
+
+  const expandConnection = async (connection: Node, predicate: Array<string>) => {
     setSelectedConnection(connection);
+    setSelectedPredicate(predicate);
+    await loadObjectPropertyDefinition();
```

Comment:
```suggestion
```
Is this line necessary? Whenever selectedPredicate does get changed it will automatically call `loadObjectPropertyDefinition()` in the useEffect